WordPress/Classes/Models/Blog.m

```diff
@@ -282,26 +286,49 @@ - (void)remove
 
         NSManagedObjectContext *context = [self managedObjectContext];
         [context deleteObject:self];
-        // For self hosted blogs, remove account unless there are other associated blogs
-        if (account && !account.isWpcom) {
-            if ([account.blogs count] == 1 && [[account.blogs anyObject] isEqual:self]) {
-                [context deleteObject:account];
-            }
-        }
+        
+        [self removeSelfHostedAccountIfNeeded:account
+                                      context:context];
+        
         [self dataSave];
         [WPAnalytics refreshMetadata];
     }];
 }
 
+/**
+ *  @brief      For self hosted blogs, removes the account unless there are other associated blogs.
+ *
+ *  @param      account     The account to remove, if it matches the criteria.  Cannot be nil.
+ *  @param      context     The context to use for the changes.  Cannot be nil.
+ */
+- (void)removeSelfHostedAccountIfNeeded:(WPAccount*)account
+                                context:(NSManagedObjectContext*)context
+{
+    NSParameterAssert([account isKindOfClass:[WPAccount class]]);
+    NSParameterAssert([context isKindOfClass:[NSManagedObjectContext class]]);
+    
+    BOOL isWpComAccount = account && !account.isWpcom;
+    
+    if (isWpComAccount) {
+        BOOL accountOnlyHasThisBlog = ([account.blogs count] == 1
+                                       && [[account.blogs anyObject] isEqual:self]);
+        
+        if (accountOnlyHasThisBlog) {
+            [context deleteObject:account];
+        }
+    }
+}
+
 - (void)setXmlrpc:(NSString *)xmlrpc
 {
     [self willChangeValueForKey:@"xmlrpc"];
     [self setPrimitiveValue:xmlrpc forKey:@"xmlrpc"];
     [self didChangeValueForKey:@"xmlrpc"];
-     _blavatarUrl = nil;
+    
+    self.blavatarUrl = nil;
 
     // Reset the api client so next time we use the new XML-RPC URL
-     _api = nil;
+    self.api = nil;
```

Comment:
Same as above, by hitting the getter we might be actually creating an instance here